src/oop/special_class_method/hard/q8/MainTest.java

```diff
@@ -0,0 +1,234 @@
+package oop.special_class_method.hard.q8;
+
+import global.ObjectTest;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class MainTest {
+    // Java
+    private final String className = "Circle";
+    private final String doubleFieldName = "radius";
+    private final String stringFieldName = "colour";
+    private final String setStringMethodName = "setColour";
+    private final String getStringMethodName = "getColour";
+    private final String getDoubleMethodName = "getRadius";
+    private final String setDoubleMethodName = "setRadius";
+    private final String testClassName = "TestClass";
+    private final double outputRadius = 3.75;
+    private final String outputColour = "Red";
+    private ObjectTest testClass;
+    private ObjectTest outputClass;
+
+    @BeforeEach
+    public void setUp() {
+        String classString = "oop.special_class_method.hard.q8." + className;
+        testClass = new ObjectTest(classString);
+        String outputClassString = "oop.special_class_method.hard.q8." + testClassName;
+        outputClass = new ObjectTest(outputClassString);
+    }
+
+    @Test
+    public void correctTestClass() throws Throwable {
+        Object classInstance = outputClass.createInstance();
+        String[] methodModifiers = {"public"};
+        Object[][] arguments = {
+                {new String[0], String[].class}
+        };
+        outputClass.callMethod("main", arguments, methodModifiers, classInstance, new Clause[]{
+                new StringLiteral("I have a " + outputRadius + " radius and my colour is " + outputColour),
+                new Optional(new StringLiteral(" ")),
+                new Optional(new NewLine())
+        });
+    }
+
+    @Test
+    public void circleClassHasCorrectAttributes() {
+        assertTrue(testClass.hasField(doubleFieldName, double.class),
+                "Your " + className + " class is missing the " + doubleFieldName + " attributes.");
```

Comment:
change attributes from plural to singular in this message and the one on line 63